src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -234,9 +234,9 @@ public Object execute(
 
     Object result = null;
 
-    updateTopologyAndConnectIfNeeded(false);
 
     try {
+      updateTopologyAndConnectIfNeeded(false);
```

Comment:
We need to remove a logic that handles closed connections inside this method (line 631)

Reply:
If the connection is closed when we call getTopology an exception will be thrown so we will never reach 631.
I am worried about the case where connection is closed but no exception is raised.